File: src/main/java/com/javahere/reachyourgoal/dto/TaskDTO.java
package com.javahere.reachyourgoal.dto;

import com.javahere.reachyourgoal.entity.TaskStatus;
import lombok.AllArgsConstructor;
import lombok.Data;

import java.time.LocalDate;

@Data
@AllArgsConstructor
public class TaskDTO {
    private String name;
    private String description;
    private TaskStatus taskStatus;
    private LocalDate taskDate;
}

File: src/main/kotlin/me/javahere/reachyourgoal/domain/dto/TaskDto.kt
package me.javahere.reachyourgoal.domain.dto

import me.javahere.reachyourgoal.domain.entity.Task

class TaskDto(
    val id: Int,
    val name: String,
    val description: String?,
) {
    fun transform(userId: Int): Task {
        return Task(
            id = id,
            name = name,
            description = description,
            userId = userId,
        )
    }
}
